From b73ceee6ffcf9edd259f2e9e7858a4f5feaeb231 Mon Sep 17 00:00:00 2001
From: Derek Foreman <derek.foreman@collabora.com>
Date: Mon, 16 Sep 2024 13:09:00 -0500
Subject: [PATCH 70/94] [PATCH 12/18] WIP: Add missing colorspace connector
 property

Gbp-Pq: Topic hdmi
Gbp-Pq: Name 0012-WIP-Add-missing-colorspace-connector-property.patch
---
 drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c | 3 ++-
 drivers/gpu/drm/drm_connector.c              | 6 ++++++
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c b/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
index b30e233da..f55e25cfd 100644
--- a/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
+++ b/drivers/gpu/drm/bridge/synopsys/dw-hdmi-qp.c
@@ -737,7 +737,8 @@ static int dw_hdmi_qp_bridge_atomic_check(struct drm_bridge *bridge,
 	if (ret)
 		dev_dbg(hdmi->dev, "%s failed: %d\n", __func__, ret);
 
-	if (old_conn_state->hdr_output_metadata != new_conn_state->hdr_output_metadata)
+	if (old_conn_state->hdr_output_metadata != new_conn_state->hdr_output_metadata ||
+	    old_conn_state->colorspace != new_conn_state->colorspace)
 		crtc_state->connectors_changed = true;
 
 	return ret;
diff --git a/drivers/gpu/drm/drm_connector.c b/drivers/gpu/drm/drm_connector.c
index 894806483..4450e1c69 100644
--- a/drivers/gpu/drm/drm_connector.c
+++ b/drivers/gpu/drm/drm_connector.c
@@ -530,6 +530,12 @@ int drmm_connector_hdmi_init(struct drm_device *dev,
 	if (max_bpc > 8)
 		drm_connector_attach_hdr_output_metadata_property(connector);
 
+        ret = drm_mode_create_hdmi_colorspace_property(connector, 0);
+        if (ret)
+                return ret;
+
+        drm_connector_attach_colorspace_property(connector);
+
 	connector->hdmi.funcs = hdmi_funcs;
 
 	return 0;
-- 
2.46.0

